qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/products/ProductsDbConnector.groovy

```diff
@@ -42,27 +35,34 @@ class ProductsDbConnector implements ListProductsDataSource {
     this.provider = Objects.requireNonNull(provider, "Provider must not be null.")
   }
 
-  @Override
+  /**
+   * Queries a data source for all available service
+   * product that have been defined by the organisation.
+   *
+   * Throws a {@link DatabaseQueryException} if the query
+   * fails for some reason. An exception must NOT be thrown,
+   * if no product can be found. The returned list needs to
+   * be empty then.
+   *
+   * @return A list of service {@link Product}.
+   * @throws DatabaseQueryException
+   */
   List<Product> findAllAvailableProducts() throws DatabaseQueryException {
     try {
-      tryToFindAllProducts()
+      def products = []
+      provider.connect().withCloseable {
+        final def query = it.prepareStatement(Queries.SELECT_ALL_PRODUCTS)
+        final ResultSet result = query.executeQuery()
+        products.addAll(convertResultSet(result))
+      }
+      return products
```

Comment:
Now it is not following Clean Clode anymore :) Would you mind changing it back?

Reply:
Why do you think this is not clean code anymore? I change `try { trySth } catch` to `try { sth } catch`

Reply:
Bob Martin explains in clean code, that try catch blocks add a level of clutter and indentation, that reduces readability. It is better, to extract code within the `try{...}` block and call a single method within it instead.